app/uk/gov/hmrc/exports/connectors/CustomsDeclarationsConnector.scala

```diff
@@ -51,7 +53,10 @@ class CustomsDeclarationsConnector @Inject() (appConfig: AppConfig, httpClient:
     }
 
   def submitCancellation(submission: Submission, xml: String)(implicit hc: HeaderCarrier): Future[String] = {
-    def actionId: String = submission.actions.find(_.requestType == SubmissionRequest).fold("Not a SubmissionRequest?")(_.id)
+    def actionId: String = submission.actions.find {
+      case _: SubmissionAction => true
+      case _                   => false
```

Comment:
What does this pattern matching provide more than the former version? :thinking:

Reply:
This is over-engineered. Why replacing a simple boolean expression with pattern matching? I'm really against this.

Reply:
If you read the comments for why we take advantage of a SubmissionAction rather than requestType == "SubmissionRequest" it might make more sense

This is just a consequence of that and the intention isn't about just replacing an if statement with pattern matching (although looking at the difference really isn't that big a deal)